ledger-core/virtual/execute/execute_test.go

```diff
@@ -488,3 +493,69 @@ func TestSMExecute_VCallResultPassedToSMObject(t *testing.T) {
 
 	mc.Finish()
 }
+
+func TestSendVStateReportWithMissingState_IfConstructorWasInterruptedBeforeRunnerCall(t *testing.T) {
+	t.Log("C5084")
+	var (
+		mc  = minimock.NewController(t)
+		ctx = context.Background()
+
+		class                              = gen.UniqueGlobalRef()
+		caller                             = gen.UniqueGlobalRef()
+		catalog     object.Catalog         = object.NewLocalCatalog()
+		authService authentication.Service = authentication.NewServiceMock(t)
+	)
+
+	slotMachine := slotdebugger.New(ctx, t)
+	slotMachine.PrepareRunner(ctx, mc)
+
+	slotMachine.AddInterfaceDependency(&catalog)
+	slotMachine.AddInterfaceDependency(&authService)
+
+	var vStateReportRecv = make(chan struct{})
+	slotMachine.PrepareMockedMessageSender(mc)
+	slotMachine.MessageSender.SendRole.Set(func(_ context.Context, msg payload.Marshaler, role node.DynamicRole, object reference.Global, pn pulse.Number, _ ...messageSender.SendOption) error {
```

Comment:
you can use `slotMachine.MessageSender.SendRole.SetCheckMessage` here

Reply:
done